radicle-node/src/runtime.rs

```diff
@@ -61,6 +62,39 @@ pub enum Error {
     GitVersion(#[from] git::VersionError),
 }
 
+/// Publishes events to subscribers.
+#[derive(Debug, Clone)]
+pub struct Emitter<T> {
+    pub(crate) subscribers: Arc<Mutex<Vec<chan::Sender<T>>>>,
+}
+
+impl<T> Default for Emitter<T> {
+    fn default() -> Emitter<T> {
+        Emitter {
+            subscribers: Default::default(),
+        }
+    }
+}
+
+impl<T: Clone> Emitter<T> {
+    /// Emit event to subscribers and drop those who can't receive it.
+    pub(crate) fn emit(&self, event: T) {
+        self.subscribers
+            .lock()
+            .unwrap()
+            .retain(|s| s.try_send(event.clone()).is_ok());
+    }
+
+    /// Subscribe to events stream.
+    pub fn subscribe(&mut self) -> chan::Receiver<T> {
+        let (sender, receiver) = chan::unbounded();
+        let mut subs = self.subscribers.lock().unwrap();
+        subs.push(sender);
+
+        receiver
+    }
```

Comment:
I added `subscribe` method on `Emitter` so that from `Service`, one can subscribe as well. (as well as from `Handler` which could come in handy for anything which interacts with `node`)